modules/etcd-overview.adoc

```diff
@@ -0,0 +1,35 @@
+// Module included in the following assemblies:
+//
+// * architecture/control-plane.adoc
+
+
+:_content-type: CONCEPT
+[id="etcd-overview_{context}"]
+= Overview of etcd
+
+etcd is a consistent, distributed key-value store that holds small amounts of data that can fit entirely in memory. Although etcd is a core component of many projects, it is the primary data store for Kubernetes, which is the standard system for container orchestration.
+
+[id="etcd-benefits_{context}"]
+== Benefits of using etcd
+
+By using etcd, you can enjoy several benefits:
+
+* Safely and automatically update containers
+* Coordinate work that is scheduled to hosts
+* Set up overlay networking for containers
+* Maintain consistent uptime for your cloud-native applications, and keep them working even if individual servers fail
+* Store and replicate all cluster states for Kubernetes
+* Distribute configuration data to provide redundancy and resiliency for the configuration of nodes
+
+[id="etcd-architecture_{context}"]
+== How etcd works
+
+To ensure a reliable approach to cluster configuration and management, the etcd Operator manages etcd within the Operator Framework. The Operator simplifies the use of etcd on a Kubernetes container platform like {product-title}. With the etcd Operator, you can create or delete etcd members, resize clusters, perform backups, and upgrade etcd.
+
+The etcd Operator observes, analyzes, and acts:
+
+. It observes the cluster state by using the Kubernetes API.
+. It analyzes differences between the current state and the state that you want.
+. It fixes the differences through the etcd cluster management APIs, the Kubernetes API, or both.
+
+Each etcd cluster member needs fast, low-latency I/O. Suitable solutions include Solid-State Drives (SSDs) on dedicated servers or fast SAN storage, such as SSD-based logical unit numbers (LUNs) on Fibre Channel (FC). Each etcd cluster member requires a storage instance. Replication is handled at the application level.
```

Comment:
while technically correct, I think it's better to leave the storage recommendations to the best practices documentation. Just to avoid conflicting information as we have to change recommendations going forward.

Reply:
Good point. I agree that we don't want to have to update recommendations in multiple places.

One of the stakeholders who was in the discussions with customers really wants us to include this information in the overview, though. (See the quote below.) I wonder if a compromise might be to remove this paragraph from the overview and replace it with wording that clearly directs the customer to the recommendations in the best practices docs, and maybe update the best practices docs a little if needed? 

This was the feedback that I received (via email) from the architect who was part of the discussion with customers:

> Hi Laura,
> 
> you correctly wrote:
> "Each etcd cluster member needs fast, low-latency I/O. Suitable solutions include Solid-State Drives (SSDs) on dedicated servers or fast SAN storage, such as SSD-based logical unit numbers (LUNs) on Fibre Channel (FC). Each etcd cluster member requires a storage instance. Replication is handled at the application level."
> 
> maybe we can hammer the message into customer's brains even more with:
> "etcd holds the cluster state, which constantly get updated. This state get continuously persisted, what leads to a high number of small changes at high frequency. That's why it is utmost important to back etcd cluster member with fast, low-latency I/O. ...."
> 
> What do you think?

Reply:
I think if we can leave it at "Each etcd cluster member needs fast, low-latency I/O. " that's enough and link to the other resource. WDYT?

Reply:
I revised the text to (hopefully) match what both you and Carsten (the other reviewer) are looking for. The updated text is:

> etcd holds the cluster state, which is constantly updated. This state is continuously persisted, what leads to a high number of small changes at high frequency. As a result, it is critical to back etcd cluster member with fast, low-latency I/O. For more information about best practices for etcd, see "Recommended etcd practices".

Reply:
Perfect!